lib/auxilium/concerns/admin_authenticated.rb

```diff
@@ -8,7 +8,7 @@ module AdminAuthenticated
       def admin_authenticated
         authenticate_user!
 
-        redirect_to root_path unless current_user.has_role?(:admin)
+        redirect_to main_app.root_path unless current_user.has_role?(:admin)
```

Comment:
Perhaps make it so, that you check for main_app, if it exists, redirect to main_app.root_path. If not: redirect to root_path?